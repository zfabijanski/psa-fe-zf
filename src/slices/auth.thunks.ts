import { createAsyncThunk } from "@reduxjs/toolkit";
import { mapAgentPositionToPositionName } from "../mapper/agent/agent";
import { api } from "../utils/api";
import { newErrorModal } from "../utils/confirmModalFactory";
import { redirect } from "../utils/router";
import {
  hideFullscreenSpinner,
  showFullscreenSpinner,
} from "slices/fullscreenSpinner";
import { init } from "../services/init";
import { showModal } from "slices/modal";
import {
  getAgentSuccess,
  signInFailure,
  signOutSuccess,
  IAuthInfo,
} from "./auth";
import { RootState } from "AppStore";

const loginUrl = "/home";

const isTrainEnv = window.location.host.includes("train.");

export const redirectToAgentPortalLandingPage = () => {
  window.location.pathname = "/home";
};

export const invalidateOtherSessions = createAsyncThunk(
  "auth/invalidateOtherSessions",
  async (_, { dispatch }) => {
    dispatch(showFullscreenSpinner());
    const config = { headers: { PSAOAuthRenew: true } };
    api
      .post("/api/auth/invalidateothersessions", {}, config)
      .then(() => {
        // @ts-expect-error
        return dispatch(init());
      })
      .then(() => redirect("/", true))
      .finally(() => {
        dispatch(hideFullscreenSpinner());
      });
  }
);

export const signOut = createAsyncThunk(
  "auth/signOut",
  async (_, { dispatch }) => {
    dispatch(
      showModal({
        modalIllustration: "padlock",
        modalTitleTrKey: "login.session.end",
        modalContentTrKey: "login.confirm.sureToLogout",
        modalButtons: {
          cancel: { textTrKey: "login.label.cancel" },
          confirm: {
            textTrKey: "login.label.logout",
            onClick: () => {
              dispatch(showFullscreenSpinner());
              api
                .delete("/api/session")
                .then(() => (window.location.href = loginUrl))
                .finally(() => {
                  dispatch(hideFullscreenSpinner());
                });
            },
          },
        },
      })
    );
  }
);

export const prolongUserSession = createAsyncThunk(
  "auth/prolongUserSession",
  () => api.get("api/auth/prolong")
);

export const getAgent = createAsyncThunk(
  "auth/getAgent",
  async (_, { dispatch }) => {
    return api
      .get<IAuthInfo>("/api/auth/agent")
      .then((agent) => ({
        ...agent,
        position: mapAgentPositionToPositionName(agent.brand, agent.position),
      }))
      .then((data) => {
        dispatch(getAgentSuccess(data));
      });
  }
);

export const redirectAndInformIfSessionExpired = createAsyncThunk(
  "auth/redirectAndInformIfSessionExpired",
  async (_, { dispatch, getState }) => {
    const state = getState() as RootState;
    redirect("/landing", true);
    if (state.auth.info) {
      dispatch(signOutSuccess());
      dispatch(
        showModal(
          newErrorModal("login.error.timeout", undefined, () => {
            window.location.href = loginUrl;
          })
        )
      );
    } else {
      window.location.href = loginUrl;
    }
  }
);

export const redirectAndInformAboutMultisession = createAsyncThunk(
  "auth/redirectAndInformAboutMultisession",
  async (_, { dispatch }) => {
    redirect("/landing", true);
    dispatch(
      showModal({
        modalContentTrKey: "login.confirm.sessionOtherDevice",
        modalButtons: {
          confirm: {
            textTrKey: "confirmWindow.yes",
            onClick: () => dispatch(invalidateOtherSessions()),
          },
          cancel: {
            textTrKey: "confirmWindow.back",
            onClick: () => redirectToAgentPortalLandingPage(),
          },
        },
      })
    );
  }
);

export const redirectAndInformAboutLoginErrors = createAsyncThunk(
  "auth/redirectAndInformAboutLoginErrors",
  async (_, { dispatch }) => {
    redirect("/landing", true);
    dispatch(signInFailure());
  }
);

export const redirectToAp = createAsyncThunk(
  "auth/redirectToAp",
  async (_, { dispatch, getState }) => {
    const { illustrations, calculator } = getState() as RootState;
    const calculationId = illustrations.previewedCalculationId;

    // Redirect to hydra without context
    let onConfirm = () => {
      window.open(`/home/api/log/hydra`, "_blank");
    };

    if (calculationId) {
      const previewedItem = illustrations.items.find(
        ({ pru_calc_calculation_id }) =>
          pru_calc_calculation_id === calculationId
      );
      const productGuid = previewedItem
        ? previewedItem.product_guid
        : calculator.current.productGuid;

      const payload = {
        productGuid,
        calculationId,
        isIllustrated: true,
      };

      // Redirect to new process in hydra
      onConfirm = () => {
        dispatch(showFullscreenSpinner());
        api
          .post<{ processInstanceId: number }>("api/calc/eapp", payload)
          .then(({ processInstanceId }) => {
            if (processInstanceId) {
              window.open(`/hydra/process/${processInstanceId}`, "_blank");
            } else {
              throw processInstanceId;
            }
          })
          .catch(() => {
            // tslint:disable-next-line: no-console
            console.warn("unable to create new process");
            dispatch(showModal(newErrorModal("auth.redirectToAp.error")));
          })
          .finally(() => {
            dispatch(hideFullscreenSpinner());
          });
      };
    }

    dispatch(
      showModal({
        modalTitleTrKey: "auth.redirectToAp.header",
        modalContentTrKey: isTrainEnv
          ? "auth.redirectToAp.trainMessage"
          : "auth.redirectToAp.message",
        modalButtons: {
          confirm: {
            textTrKey: "modal.ok",
            onClick: onConfirm,
          },
        },
      })
    );
  }
);
