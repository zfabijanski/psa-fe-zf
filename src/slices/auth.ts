import { createSlice, createSelector, PayloadAction } from "@reduxjs/toolkit";

const loginUrl = "/home";

export const redirectToAgentPortalLandingPage = () => {
  window.location.pathname = loginUrl;
};

export enum Brand {
  BrandPP = "BRAND_PP",
}

export interface IAuthInfo {
  login: string;
  givenName: string;
  surname: string;
  brand: Brand;
  agentNo: string;
  phoneNumber: string;
  email: string;
  roles: string[];
  position: string;
  picturePath?: string;
  apLink?: string;
}

export type AuthState = {
  info?: IAuthInfo;
  loginFailed: boolean;
};

export const initialState: AuthState = {
  loginFailed: false,
};

export const authSlice = createSlice({
  name: "auth",
  initialState,
  reducers: {
    getAgentSuccess(state, action: PayloadAction<IAuthInfo>) {
      state.info = action.payload;
      state.loginFailed = false;
    },
    signOutSuccess(state) {
      state.info = undefined;
    },
    signInFailure(state) {
      state.loginFailed = true;
    },
  },
});

export const getCurrentAgentNo = createSelector(
  (state: AuthState) => state.info,
  (info) => info && info.agentNo
);

export const { getAgentSuccess, signOutSuccess, signInFailure } =
  authSlice.actions;

export * from "./auth.thunks";

// @TODO REMOVE
export enum LoginActionType {
  GetAgentSuccess = "auth/getAgentSuccess",
  SignOutSuccess = "auth/signOutSuccess",
  SignInFailure = "auth/signInFailure",
}
