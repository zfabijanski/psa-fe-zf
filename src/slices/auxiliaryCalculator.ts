import { createSlice, PayloadAction } from "@reduxjs/toolkit";

const initialState = {
  isOpen: false,
};

export const auxiliaryCalculatorSlice = createSlice({
  name: "auxiliaryCalculator",
  initialState,
  reducers: {
    setOpen(state, action: PayloadAction<boolean>) {
      state.isOpen = action.payload;
    },
  },
});

export const { setOpen } = auxiliaryCalculatorSlice.actions;
