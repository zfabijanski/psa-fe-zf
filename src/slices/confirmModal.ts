import { createSlice, PayloadAction, createSelector } from "@reduxjs/toolkit";
import { ReactNode } from "react";
import { MessageDescriptor } from "react-intl";
import { TranslationValue } from "utils/types";

export enum Variant {
  Confirm = "Confirm",
  Deny = "Deny",
}

export interface IConfirmModalOptions {
  variant?: Variant;
  header?: string | MessageDescriptor;
  message?: string | MessageDescriptor;
  messageValues?: Record<string, TranslationValue>;
  onConfirm?: () => void;
  onCancel?: () => void;
  confirmText?: string | MessageDescriptor;
  showBackdrop?: boolean;
  showCancel?: boolean;
  showConfirm?: boolean;
  children?: ReactNode;
  formattedText?: string;
  className?: string;
}

type State = {
  options: IConfirmModalOptions | null;
};

const initialState: State = {
  options: null,
};

export const confirmModalSlice = createSlice({
  name: "confirmModal",
  initialState,
  reducers: {
    openConfirmModal(state, action: PayloadAction<IConfirmModalOptions>) {
      const defaultOptions = {
        confirmText: { id: "confirmWindow.yes" },
        showBackdrop: true,
        showCancel: true,
        showConfirm: true,
        variant: Variant.Confirm,
      };

      state.options = {
        ...defaultOptions,
        ...action.payload,
      };
    },
    closeConfirmModal(state) {
      state.options = null;
    },
  },
});

export const { openConfirmModal, closeConfirmModal } =
  confirmModalSlice.actions;

export const isConfirmModalOpen = createSelector(
  [(state: { confirmModal: State }) => state.confirmModal],
  (modalState) => !!modalState.options
);
