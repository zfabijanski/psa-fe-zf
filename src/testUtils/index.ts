export * from "./TestProviders";
export * from "./mockCreators";
export * from "./helpers";
