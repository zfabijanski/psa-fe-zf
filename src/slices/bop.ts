import {
  CancelStatus,
  IQuestionJSON,
  ParsedAnswers,
  QuestionDataTypeCode,
  SurveyType,
} from "common/features/Survey/Survey.types";
import {
  generateSurveyActions,
  generateSurveyActionType,
  generateSurveyStore,
  generateSurveyThunks,
  GenerateSurveyThunksConfig,
} from "common/features/Survey/store";
import { showModal } from "./modal";
import { ModalTypes } from "models/Modal";
import {
  hideFullscreenSpinner,
  showFullscreenSpinner,
} from "./fullscreenSpinner";
import { getCurrentMeetingId, setCurrentMeetingOutdated } from "./meetings";
import { getCurrentAgentNo } from "./auth";
import { api } from "utils/api";
import { redirect } from "utils/router";

export interface IQuestionRecord {
  data_type: QuestionDataTypeCode;
  answer_type_code: string;
  question_value: string;
  answerStatus: ParsedAnswers[];
  cancelStatus: CancelStatus;
  prompt_value: string;
  tooltip: string;
  code: string;
  type: string;
}

export enum BopModalStatus {
  Refuse = "refuse",
  Inform = "inform",
  End = "end",
  Error = "error",
}

export interface IBOP {
  questions: IQuestionRecord[];
  adequacy_id: number | null;
  isQuestionLoading: boolean;
  isDatePickerOpened: boolean;
  isQuestionDeleting: boolean;
  allSteps: number;
  currentStep: number;
}

export interface IQuestionResponse {
  adequacy_id: number;
  adequacy_version: string;
  step_all_no: number;
  step_no: number;
  questionnaire: IQuestionJSON[];
  questionnaire_end: "Y" | "N";
}

export const BopActionType = generateSurveyActionType(SurveyType.bop);

export const actions = generateSurveyActions(SurveyType.bop);
const {
  addNextQuestion,
  deleteApproval,
  deleteLastQuestion,
  markAsApproved,
  resetAnswers,
  resetQuestionnaire,
  setAdequacyID,
  setQuestionDeletingSpinner,
  setQuestionnaireProgress,
  setSpinner,
  toggleAnswerRefusal,
  updateMultiSelectAnswer,
  updateNestedMultiSelectAnswer,
  updateNestedSingleSelectAnswer,
  updateSingleSelectAnswer,
  updateStringAnswer,
} = actions;
export {
  addNextQuestion,
  deleteApproval,
  deleteLastQuestion,
  markAsApproved,
  resetAnswers,
  resetQuestionnaire,
  setAdequacyID,
  setQuestionDeletingSpinner,
  setQuestionnaireProgress,
  setSpinner,
  toggleAnswerRefusal,
  updateMultiSelectAnswer,
  updateNestedMultiSelectAnswer,
  updateNestedSingleSelectAnswer,
  updateSingleSelectAnswer,
  updateStringAnswer,
};

const bopSlice = generateSurveyStore(BopActionType);
export { bopSlice };

const onStopQuestionnaireConfirm: GenerateSurveyThunksConfig["handlers"]["onStopQuestionnaireConfirm"] =
  (dispatch, getState, { getUrl, handleApiErrorFactory }) =>
    dispatch(
      showModal({
        modalIllustration: "alert-double-triangle",
        modalTitleTrKey: "bop.confirm.denyTitle",
        modalType: ModalTypes.danger,
        modalContentTrKey: "bop.message.refuse",
        modalButtons: {
          confirm: {
            // TODO: apply deny styling?
            textTrKey: "confirmWindow.ok",
            onClick: () => {
              dispatch(showFullscreenSpinner());
              const meetingId = getCurrentMeetingId(getState().meetings);
              const agentNo = getCurrentAgentNo(getState().auth);
              const adequacyId = getState().bop.adequacy_id;
              api
                .post(getUrl("meeting-denied"), {
                  i_agent_no: agentNo,
                  i_application_id: "psa-be-main",
                  i_meeting_id: meetingId,
                  i_source_system: "PSAONLINE",
                  io_adequacy_id: adequacyId === 0 ? null : adequacyId,
                })
                .then(() => {
                  dispatch(resetQuestionnaire());
                  dispatch(setCurrentMeetingOutdated());
                  redirect("/meeting");
                })
                .catch(handleApiErrorFactory(dispatch, resetQuestionnaire))
                .finally(() => {
                  dispatch(hideFullscreenSpinner());
                });
            },
          },
        },
      })
    );

const onStopQuestionnaireCancel: GenerateSurveyThunksConfig["handlers"]["onStopQuestionnaireCancel"] =
  (dispatch, _, { shouldToggleCheckbox }) =>
    shouldToggleCheckbox && dispatch(toggleAnswerRefusal());

const {
  getNextQuestion,
  refuseAnswer,
  singleSelectClickHandler,
  startQuestionnaire,
  stopQuestionnaire,
  undoLastQuestion,
  isValid,
} = generateSurveyThunks({
  type: SurveyType.bop,
  actions,
  handlers: {
    onStopQuestionnaireConfirm,
    onStopQuestionnaireCancel,
  },
});

export {
  getNextQuestion,
  refuseAnswer,
  singleSelectClickHandler,
  startQuestionnaire,
  stopQuestionnaire,
  undoLastQuestion,
  isValid,
};
