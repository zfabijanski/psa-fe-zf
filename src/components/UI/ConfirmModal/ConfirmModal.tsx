import React from "react";
import { FocusOn } from "react-focus-on";
import { FormattedMessage, MessageDescriptor } from "react-intl";
import styled from "styled-components";
import { IConfirmModalOptions, Variant } from "slices/confirmModal";
import { ColorType } from "../../../theme/types";
import { useI18n } from "../../../utils/i18n";
import { Backdrop } from "../Backdrop";
import PruButton from "../PruButton/PruButton";
import PruIcon from "../PruIcon/PruIcon";
import ScrollableContainer from "../ScrollableContainer/ScrollableContainer";
import { Box, Flex } from "../Box";

const Modal = styled.div.attrs({
  role: "dialog",
})`
  width: 652px;
  position: fixed;
  z-index: 400;
  left: calc(50% - 326px);
  top: 30%;
  @media (max-height: 768px) {
    top: 20%;
  }
`;

const Content = styled.div`
  width: 100%;
  border-radius: 7px;
  -webkit-backdrop-filter: blur(10px);
  backdrop-filter: blur(10px);
  background-color: #ffffff;
  padding: 30px;
  margin-bottom: 20px;

  h2 {
    height: 28px;
    font-size: 26px;
    font-weight: normal;
    font-style: normal;
    font-stretch: normal;
    line-height: 1.08;
    text-align: center;
  }

  p {
    font-size: 18px;
    font-stretch: normal;
    line-height: 1.33;
    text-align: center;
    color: ${({ theme }) => theme.newColors.gray100};
    margin: 0;
  }

  div {
    font-size: 18px;
    font-stretch: normal;
    line-height: 1.33;
    margin: 0;
  }
`;

const Message = styled.div`
  display: flex;
  flex-direction: column;
`;

const SuccessButtonContainer = styled.div`
  text-align: center;
`;

const PrevButton = styled(PruButton)`
  text-align: right;
  padding: 0;
  min-width: 187px;
  outline: none;
  box-shadow: none;
`;

const PrevIcon = styled(PruIcon)`
  display: inline;
  margin-right: 6px;
  > img {
    margin-bottom: 4px;
  }
`;
const Column = styled(Box)({
  width: "33.33%",
});

const labelBack: MessageDescriptor = { id: "confirmWindow.back" };

export const ConfirmModal: React.FC<IConfirmModalOptions> = ({
  header,
  message,
  messageValues,
  confirmText,
  onCancel,
  onConfirm,
  showBackdrop = true,
  showCancel,
  showConfirm = true,
  variant,
  children,
  formattedText,
  className,
}) => {
  const i18n = useI18n();

  const handleConfirmClick = () => {
    if (onConfirm) {
      onConfirm();
    }
  };

  const getVariantStyling = () => {
    switch (variant) {
      case Variant.Deny:
        return {
          backgroundColor: "red" as ColorType,
          color: "white" as ColorType,
        };
      default:
        return {
          backgroundColor: "green" as ColorType,
          color: "white" as ColorType,
        };
    }
  };

  return (
    <div className={className}>
      <FocusOn enabled={showBackdrop}>
        {showBackdrop && <Backdrop />}
        <Modal className="modal">
          <Content>
            <Message className={"message"}>
              <ScrollableContainer overflowY={null}>
                {header && <h2>{i18n.messageToString(header)}</h2>}
                {message && (
                  <p>{i18n.messageToString(message, messageValues)}</p>
                )}
                {formattedText && (
                  <div dangerouslySetInnerHTML={{ __html: formattedText }} />
                )}
              </ScrollableContainer>
              {children}
            </Message>
          </Content>
          <Flex>
            <Column>
              {showCancel && (
                <PrevButton
                  buttonType="secondary"
                  children={
                    <div>
                      <PrevIcon
                        type={
                          showBackdrop ? "arrow-prev-white" : "arrow-prev-black"
                        }
                        alt={i18n.messageToString(labelBack) as string}
                      />
                      <FormattedMessage id={labelBack.id} />
                    </div>
                  }
                  onClick={onCancel}
                />
              )}
            </Column>
            <Column>
              {showConfirm && (
                <SuccessButtonContainer>
                  <PruButton
                    {...getVariantStyling()}
                    onClick={handleConfirmClick}
                  >
                    {i18n.messageToString(confirmText)}
                  </PruButton>
                </SuccessButtonContainer>
              )}
            </Column>
            <Column />
          </Flex>
        </Modal>
      </FocusOn>
    </div>
  );
};

export default ConfirmModal;
