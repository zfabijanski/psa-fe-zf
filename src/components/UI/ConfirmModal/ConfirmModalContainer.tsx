import { connect } from "react-redux";
import { Dispatch } from "@reduxjs/toolkit";

import throttle from "lodash/throttle";

import { closeConfirmModal, IConfirmModalOptions } from "slices/confirmModal";
import { RootState } from "../../../AppStore";
import ConfirmModal from "./ConfirmModal";

const mapStateToProps = ({ confirmModal }: RootState) => ({
  options: confirmModal.options,
});

interface IConfirmModalProps {
  options: IConfirmModalOptions | null;
  dispatch: Dispatch;
}

const ConfirmModalContainer = ({ options, dispatch }: IConfirmModalProps) => {
  const handleConfirm = throttle(
    () => {
      dispatch(closeConfirmModal());
      if (options && options.onConfirm) {
        options.onConfirm();
      }
    },
    2000,
    { leading: true, trailing: false }
  );

  const handleCancel = throttle(
    () => {
      dispatch(closeConfirmModal());
      if (options && options.onCancel) {
        options.onCancel();
      }
    },
    2000,
    { leading: true, trailing: false }
  );

  return (
    <>
      {!!options && (
        <ConfirmModal
          {...options}
          onConfirm={handleConfirm}
          onCancel={handleCancel}
        />
      )}
    </>
  );
};

export default connect(mapStateToProps)(ConfirmModalContainer);
