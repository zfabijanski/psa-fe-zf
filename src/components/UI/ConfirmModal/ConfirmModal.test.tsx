import { IConfirmModalOptions, Variant } from "slices/confirmModal";
import { ConfirmModal } from "./ConfirmModal";
import { render, screen } from "testUtils";

const mockOnConfirm = jest.fn();
const mockOnCancel = jest.fn();

const defaultProps: IConfirmModalOptions = {
  confirmText: "Confirm",
  onConfirm: mockOnConfirm,
  onCancel: mockOnCancel,
};

const setup = (props: Partial<IConfirmModalOptions> = {}) =>
  render(<ConfirmModal {...defaultProps} {...props} />);

describe("ConfirmModal", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it("should call onConfirm when confirm button is clicked", () => {
    setup();
    screen.getByText("Confirm").click();
    expect(mockOnConfirm).toHaveBeenCalledTimes(1);
  });

  it("should call onCancel when cancel button is clicked", () => {
    setup({ showCancel: true });
    screen.getByText("confirmWindow.back").click();
    expect(mockOnCancel).toHaveBeenCalledTimes(1);
  });

  it("should handle different variants", () => {
    setup({ variant: Variant.Deny });
    expect(screen.getByText("Confirm")).toHaveStyle(
      "background-color: rgb(5, 90, 96)"
    );
  });

  it("should render the header", () => {
    setup({ header: "header" });
    expect(screen.getByText("header")).toBeInTheDocument();
  });

  it("should render the message", () => {
    setup({ message: "message" });
    expect(screen.getByText("message")).toBeInTheDocument();
  });

  it("should render the formatted text", () => {
    setup({ formattedText: "<p>formatted text</p>" });
    expect(screen.getByText("formatted text")).toBeInTheDocument();
  });

  it("should change icon when showBackdrop is false", () => {
    setup({ showCancel: true, showBackdrop: false });
    expect(screen.getByText("confirmWindow.back")).toMatchSnapshot();
  });
});
