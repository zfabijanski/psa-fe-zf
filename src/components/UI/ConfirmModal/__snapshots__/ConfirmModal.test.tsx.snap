// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal should change icon when showBackdrop is false 1`] = `
.c0 {
  display: inline;
  margin-right: 6px;
}

.c0 > img {
  margin-bottom: 4px;
}

@media (max-height:768px) {

}

<div>
  <div
    alt="confirmWindow.back"
    class="c0"
    type="arrow-prev-black"
  >
    <img
      alt="confirmWindow.back"
      src="arrow-prev-black.svg"
    />
  </div>
  confirmWindow.back
</div>
`;

exports[`ConfirmModal should render correctly 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  position: fixed;
  z-index: 300;
  left: 0;
  top: 0;
  -webkit-backdrop-filter: blur(10px);
  -webkit-backdrop-filter: blur(10px);
  backdrop-filter: blur(10px);
  background-color: rgba(0,0,0,0.7);
}

.c8 {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0px;
  cursor: pointer;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 128px;
  max-width: 400px;
  height: 52px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 16px;
  font-weight: 700;
  line-height: 20px;
  text-align: center;
  color: #FFFFFF;
  background-color: #055A60;
  border: 2px solid #055A60;
}

.c8 svg g,
.c8 svg g path,
.c8 svg g circle,
.c8 svg g rect {
  stroke: #FFFFFF;
}

.c8 svg g rect {
  fill: #FFFFFF;
}

.c8 svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c8 svg > path {
  stroke: #FFFFFF;
}

.c8:hover {
  color: #FFFFFF;
  background-color: #181818;
  border: 2px solid #181818;
}

.c8:hover svg g,
.c8:hover svg g path,
.c8:hover svg g circle,
.c8:hover svg g rect {
  stroke: #FFFFFF;
}

.c8:hover svg g rect {
  fill: #FFFFFF;
}

.c8:hover svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c8:hover svg > path {
  stroke: #FFFFFF;
}

.c8:disabled {
  cursor: initial;
  color: #FFFFFF;
  background-color: #9BBDBF;
  border: 2px solid #9BBDBF;
}

.c8:disabled svg g,
.c8:disabled svg g path,
.c8:disabled svg g circle,
.c8:disabled svg g rect {
  stroke: #FFFFFF;
}

.c8:disabled svg g rect {
  fill: #FFFFFF;
}

.c8:disabled svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c8:disabled svg > path {
  stroke: #FFFFFF;
}

.c8:active:not(:disabled) {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

.c4 {
  height: 100%;
  width: 100%;
  -webkit-scrollbar-color: #9b9b9b transparent;
  -moz-scrollbar-color: #9b9b9b transparent;
  -ms-scrollbar-color: #9b9b9b transparent;
  scrollbar-color: #9b9b9b transparent;
  -webkit-scrollbar-width: thin;
  -moz-scrollbar-width: thin;
  -ms-scrollbar-width: thin;
  scrollbar-width: thin;
  -webkit-scrollbar-height: thin;
  -moz-scrollbar-height: thin;
  -ms-scrollbar-height: thin;
  scrollbar-height: thin;
}

.c4::-webkit-scrollbar-corner {
  background: transparent;
}

.c4::-webkit-scrollbar-track {
  background: transparent;
}

.c4::-webkit-scrollbar-thumb {
  background: #9b9b9b;
}

.c4::-webkit-scrollbar {
  width: 6px;
}

.c4::-webkit-scrollbar {
  height: 6px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  width: 652px;
  position: fixed;
  z-index: 400;
  left: calc(50% - 326px);
  top: 30%;
}

.c2 {
  width: 100%;
  border-radius: 7px;
  -webkit-backdrop-filter: blur(10px);
  -webkit-backdrop-filter: blur(10px);
  backdrop-filter: blur(10px);
  background-color: #ffffff;
  padding: 30px;
  margin-bottom: 20px;
}

.c2 h2 {
  height: 28px;
  font-size: 26px;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.08;
  text-align: center;
}

.c2 p {
  font-size: 18px;
  font-stretch: normal;
  line-height: 1.33;
  text-align: center;
  color: #2F2F2F;
  margin: 0;
}

.c2 div {
  font-size: 18px;
  font-stretch: normal;
  line-height: 1.33;
  margin: 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  text-align: center;
}

.c6 {
  width: 33.33%;
}

@media (max-height:768px) {
  .c1 {
    top: 20%;
  }
}

<div>
  <div>
    <div
      aria-hidden="true"
      data-focus-guard="true"
      data-focus-on-hidden="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        aria-label="backdrop"
        class="c0"
      />
      <div
        class="c1 modal"
        role="dialog"
      >
        <div
          class="c2"
        >
          <div
            class="c3 message"
          >
            <div
              class="c4"
            />
          </div>
        </div>
        <div
          class="c5"
        >
          <div
            class="c6"
          />
          <div
            class="c6"
          >
            <div
              class="c7"
            >
              <button
                class="c8"
                color="white"
                type="button"
              >
                Confirm
              </button>
            </div>
          </div>
          <div
            class="c6"
          />
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      data-focus-guard="true"
      data-focus-on-hidden="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
  <div
    aria-hidden="true"
    data-focus-on-hidden="true"
  />
</div>
`;
